crates/sui-types/src/move_package.rs

```diff
@@ -154,9 +242,82 @@ impl MovePackage {
                 })
                 .collect(),
             max_move_package_size,
+            type_origin_table,
+            linkage_table,
         )
     }
 
+    fn build_linkage_table(
+        dependencies: &[MovePackage],
+    ) -> Result<BTreeMap<ObjectID, UpgradeInfo>, ExecutionError> {
+        // all transitive dependencies are included so simply put them into the table
+        let mut linkage_table = BTreeMap::new();
```

Comment:
I think you could probably use `BTreeMap::from_iter` like you've done below now, because there are no `Err`or cases now, when building the linkage table.